### DISCUSSION TITLE
Are Lighthouse decks known to be fragile and break often?

### DISCUSSION THREAD

**Discussion by mstroia** [bitcraze/discussions#921] [Q&A]
I am currently working on a Crazyflie 2.X using a lighthouse deck for positioning. Quite suddenly, the lighthouse deck could no longer get information from the lighthouses reliably, and it can no longer calibrate. (When calibrating I get a: No sweep recorded! message). 

A while back, a person working on the same project experienced similar issues, issues which were solved with replacing the board for a new one. So my question is, are the boards fragile and prone to break, or is there something that might cause this? Is there a way to quickly determine if the issue is in the hardware and it requires replacement, or in the software?

**Comment by tobbeanton** [comment]
The 4 x optical sensors on the edges of the board can be a bit fragile and sometime break off during crashes. Also the deck connectors can wear out. If you are not changing decks I recommend soldering the pins to the deck as described in [this](https://www.bitcraze.io/documentation/tutorials/solder-deck-pins/) guide.

In your case this might be a software bug though. Can you plot the log variable [rawAngle0xlh2](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#lighthouserawangle0xlh2) to see if you receive sweep angles?

**Reply by mstroia** [reply]
Soldering may indeed be an option (for a working board), though it might still be of interest to see if the issue is a software bug or not. Here's the plot for rawAngle variable (nothing seems to be changing in the plot)![image](https://github.com/bitcraze/discussions/assets/36758927/a3774ff8-6cf5-452c-b5f6-53a093ed9ccd)

**Reply by tobbeanton** [reply]
Did you move around the CF without the angles changes? Else it will not change much. This is what I get:
![image](https://github.com/bitcraze/discussions/assets/6593186/64c21653-a33b-4437-8722-8123f201fb7d)

When I try estimating the geometry I get the same thing, no sweep angles, I'm smelling a bug...

**Reply by tobbeanton** [reply]
I moved the CF a bit and now I'm getting sweep angles.

**Reply by mstroia** [reply]
Surprisingly, if I shook the CF around a bit when it didn't detect the sweeps it ended up working. I can't say if this is for sure the reason for which it didn't work, but I didn't change that much in my approach besides that.

**Reply by mstroia** [reply]
As an addition, after a while I couldn't get the sweeps to be detected again. When I restarted the cfclient, the sweeps were detected. The hardware was the same, the environment was the same, the only thing that different was that the client was recently restarted. Is there anything that could cause this, or was it just a coincidence? (I have tried detecting the sweeps for 5+ minutes before the restart happened)

**Reply by knmcguire** [reply]
Hmmm... we haven't see this issue before with the cfclient that the plot value freezes or stays the same. I would argue that the sweepangle detection and the restart is just a coincidence... because the cfclient should have 0 influence on that. 

By the way, have you checked the [console tab](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/console_tab/) of the cfclient as well? there were no errors indicated related to the lighthouse deck?

**Reply by mstroia** [reply]
I also expected the client to have nothing to do with this, that's why I considered the idea of a coincidence in the first place.

There was no error in the console tab, everything seems to work fine otherwise. If the CF was previously calibrated, it could even fly. The only thing that didn't work is trying to detect the sweep angles again.I got it working for now, but it might be of interest to find out why the error happens so it can be avoided in the future.

**Reply by knmcguire** [reply]
is it perhaps that maybe something is interfering when you tried calibrating it? Are you using 2 or 4 basestations and are there many reflective objects like windows or mirrors in the area?
